basic-cache/basic-cache-proto/proto/v1/basic-cache.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package blockscout.basicCache.v1;
+
+option go_package = "github.com/blockscout/blockscout-rs/basic-cache";
+
+import "v1/convert_options.proto";
+
+service Cache {
+  rpc CreateSmartContract(CreateSmartContractRequest) returns (SmartContract) {}  
+  rpc GetSmartContract(GetSmartContractRequest) returns (SmartContract) {}
+}
+
+// could not use struct for id because `actix-prost` cannot work with setting
+// such parameters from query :((
+//
+// message SmartContractId {
+//   string chain_id = 1;
+//   string address = 2;
+// }
+
+message SourceFile {
+  string name = 1;
+  string content = 2;
+}
+
+message SmartContract {
+  string url = 1
+    [ (convert_options.convert) = {type: "url::Url"} ];
+  repeated SourceFile sources = 2;
```

Comment:
why not HashMap to prevent `name` duplicates

Reply:
With `map` sending duplicate filenames is silently handled and one of the files is dropped. It can't be checked the same way as it was with `repeated`, so I can't throw an error during validation.

Won't `repeated` be better in this case?